### PR TITLE
EN6-129 fix resources filters and sorting

### PR DESCRIPTION
**Pull request by ffleandro**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by joewhite101** [comment]
Retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceDAO.java [hunk 1]
```diff
@@ -300,24 +300,17 @@ private PreparedStatement buildStatement(FieldSearchFilter[] filters, List<Strin
...
         super.appendOrderQueryBlocks(filters, query, false);
         return query.toString();
     }
 
     private StringBuffer createBaseQueryBlock(FieldSearchFilter[] filters, boolean selectAll, List<String> categories) {
         StringBuffer query = super.createBaseQueryBlock(filters, selectAll);
-        if (null != categories && categories.size() > 0) {
-            query.append("INNER JOIN resourcerelations ON resources.resid = resourcerelations.resid ");
+
+        for(int i = 0; i < categories.size(); i++) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceDAO.java]: Categories can be null here based on the old code.

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java [hunk 2]
```diff
@@ -307,8 +307,8 @@ private EntitySearchFilter createOrderFilter(RestListRequest requestList) {
             key = IResourceManager.RESOURCE_FILENAME_FILTER_KEY;
         }
 
-        EntitySearchFilter filter = new EntitySearchFilter(key, false);
-        filter.setOrder(requestList.getDirection().equals("desc") ?
+        EntitySearchFilter filter = new EntitySearchFilter(key, true);
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java]: What does this change do?

Comment (ffleandro) [reply plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java]: From the docs:
`Decide whether the filter is applied to an Entity Attribute (true) or to a metadata (false)`

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by pietrangelo** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by pietrangelo** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

**Pull request merged**